msal/src/main/java/com/microsoft/identity/client/statemachine/states/SignInStates.kt

```diff
@@ -266,7 +266,7 @@ class SignInPasswordRequiredState(
      * @param callback [com.microsoft.identity.client.statemachine.states.SignInPasswordRequiredState.SubmitPasswordCallback] to receive the result on.
      * @return The results of the submit password action.
      */
-    fun submitPassword(password: String, callback: SubmitPasswordCallback) {
+    fun submitPassword(password: CharArray, callback: SubmitPasswordCallback) {
         LogSession.logMethodCall(TAG, "${TAG}.submitPassword")
         NativeAuthPublicClientApplication.pcaScope.launch {
             try {
```

Comment:
Not sure exactly where we need to add StringUtil.overwriteWithZero(params.password)

Every place password is passed as param? If so, then we need it here as well i think

Reply:
Maybe right after we get the result

Reply:
This password is coming from the calling application and it must be their responsibility to clear the password char array. As a rule, I am clearing password char arrays in the object we are creating.